### PR TITLE
Add ExternalSource to Planner Task, Plan and Bucket

### PR DESCRIPTION
**Pull request by danlucuiPlanner**

### CODE REVIEW

File: api-reference/beta/resources/plannertaskcreation.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ Contains information about the origin of the [plannerTask](plannerTask.md). This
 |Property|Type|Description|
 |:---|:---|:---|
 |teamsPublicationInfo|[plannerTeamsPublicationInfo](../resources/plannerteamspublicationinfo.md)|Information about the publication process that created this task. `null` value indicates that the task was not created by a publication process.|
+|creationSourceKind|plannerCreationSourceKind|Specifies what kind of creation source the task is created with. |
```
Comment (dodaromike) [inline api-reference/beta/resources/plannertaskcreation.md:20]: Please add the list of possible values following the guidelines here: https://dev.azure.com/msazure/One/_wiki/wikis/Microsoft%20Graph%20Partners/211747/Enumerations?anchor=examples

File: api-reference/beta/resources/plannertaskcreation.md [hunk 2]
```diff
@@ -17,6 +17,7 @@ Contains information about the origin of the [plannerTask](plannerTask.md). This
 |Property|Type|Description|
```
Comment (isvargasmsft) [inline api-reference/beta/resources/plannertaskcreation.md:17]: Line 15: Please add a note indicating that this entity is base type of plannerTeamsPublicationInfo resource. And also, add a similar note to the plannerTeamsPublicationInfo topic, indicating that the resource now inherits from plannerTaskCreation, and add the inherited properties to the Properties table, indicating that those are inherited from plannerTaskCreation.

File: api-reference/beta/resources/plannerexternaltasksource.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+ms.prod: "planner"
+doc_type: resourcePageType
+---
+
+# plannerExternalTaskSource resource type
+
+Namespace: microsoft.graph
+
+Contains information about the relationship of a [plannerTask](plannerTask.md) to a user experience outside of Planner. Tasks in Planner can be surfaced in or synced with other experiences to track work in the context of that experience. Data in the plannerExternalTaskSource can be displayed in a user interface, used for sync information utilized by external services, or used to simply point to where a task was created in an external service.
+
```
Comment (dodaromike) [inline api-reference/beta/resources/plannerexternaltasksource.md:15]: Please add note that indicates this type is derived from plannerTaskCreation.  Also add the inherited properties to the properties table.

File: api-reference/beta/resources/plannerplancreation.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+ms.localizationpriority: medium
+ms.prod: "planner"
+doc_type: resourcePageType
+---
+
+# plannerPlanCreation resource type
+
+Namespace: microsoft.graph
+
+The resources that derive from plannerPlanCreation contains information about the origin of the [plannerPlan](plannerplan.md). Apps do not need to know the origin of the plan to be able to work with it; however, some apps can use the additional information to provide specific experiences around these plans.
```
Comment (dodaromike) [inline api-reference/beta/resources/plannerplancreation.md:14]:
```suggestion
The resources that derive from plannerPlanCreation contain information about the origin of the [plannerPlan](plannerplan.md). Apps do not need to know the origin of the plan to be able to work with it; however, some apps can use the additional information to provide specific experiences around these plans.
```

File: api-reference/beta/resources/plannerplancreation.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+## Properties
+|Property|Type|Description|
+|:---|:---|:---|
+|creationSourceKind|plannerCreationSourceKind|Specifies what kind of creation source the plan is created with.|
```
Comment (dodaromike) [inline api-reference/beta/resources/plannerplancreation.md:19]: Please include a list of possible values of the enum in the description.

File: api-reference/beta/resources/plannerbucketcreation.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+ms.localizationpriority: medium
+ms.prod: "planner"
+doc_type: resourcePageType
+---
+
+# plannerBucketCreation resource type
+
+Namespace: microsoft.graph
+
+The resources that derive from plannerBucketCreation contains information about the origin of the [plannerBucket](plannerbucket.md). Apps do not need to know the origin of the bucket to be able to work with it; however, some apps can use the additional information to provide specific experiences around these buckets.
```
Comment (dodaromike) [inline api-reference/beta/resources/plannerbucketcreation.md:14]:
```suggestion
The resources that derive from plannerBucketCreation contain information about the origin of the [plannerBucket](plannerbucket.md). Apps do not need to know the origin of the bucket to be able to work with it; however, some apps can use the additional information to provide specific experiences around these buckets.
```

**Review by dodaromike — CHANGES_REQUESTED** [review]
Looks good but please review comments and add a few things.

File: api-reference/beta/resources/plannerexternaltasksource.md (hunk 3 above)
Comment (victorliu-microsoft) [reply api-reference/beta/resources/plannerexternaltasksource.md:15]: should the note be in the Relationships section? or is it enough to put it in the description?

Comment (dodaromike) [reply api-reference/beta/resources/plannerexternaltasksource.md:15]: Seems best to put it in the description where all users will be looking.

**Review by dodaromike — APPROVED** [review]
Thanks for changes.  Sorry about the delay.

File: api-reference/beta/resources/plannerexternalbucketsource.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+Namespace: microsoft.graph
+
+Contains information about the relationship of a [plannerBucket](plannerbucket.md) to a user experience outside of Planner. Buckets in Planner can be surfaced in or synced with other experiences to track work in the context of that experience. Data in the plannerExternalBucketSource can be displayed in a user interface, used for sync information utilized by external services, or used to simply point to where a bucket was created in an external service. This type derives from [plannerBucketCreation](plannerBucketCreation.md).
+
+## Properties
+|Property|Type|Description|
+|:---|:---|:---|
+|creationSourceKind|plannerCreationSourceKind|Specifies what kind of creation source the bucket is created with. The possible values are: `external`, `publication`. Inherited from [plannerBucketCreation](plannerBucketCreation.md).|
+|contextScenarioId|String| Nullable. An identifier for the scenario associated with this external source. This will generally take the form of "com.contoso.myapp".|
```
Comment (TarkanSevilmis) [inline api-reference/beta/resources/plannerexternalbucketsource.md:20]: This should mention the reverse dns format is recommended. The sample and "generally take form of" isn't informative enough. We can this to be self applied by people as a guideline.

File: api-reference/beta/resources/plannerexternalbucketsource.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+|externalObjectId|String| Nullable. The id of the entity that an external service associates with a bucket.|
```
Comment (TarkanSevilmis) [inline api-reference/beta/resources/plannerexternalbucketsource.md:21]: We need to mention somewhere contextscenarioId + externalobject id is enforced to be unique.

**Comment by victorliu-microsoft** [comment]
@isvargasmsft @dodaromike how did the changelog links suddenly fail?

**Review by dodaromike — APPROVED** [review]
Looks fine.  I have not reviewed the changelog links.

**Review by TarkanSevilmis — APPROVED** [review]
Looks technically acccurate.

**Review by isvargasmsft — APPROVED** [review]

**Pull request merged**